com.unity.render-pipelines.core/Editor/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,181 @@
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+namespace UnityEditor.Rendering
+{
+	/// <summary>
+	/// Material upgrader and relevant utilities for SpeedTree8.
+	/// </summary>
+	public class SpeedTree8MaterialUpgrader : MaterialUpgrader
+    {
+        private enum WindQuality
+        {
+            None = 0,
+            Fastest,
+            Fast,
+            Better,
+            Best,
+            Palm,
+            Count
+        }
+
+        private static string[] WindQualityString =
+        {
+            "_WINDQUALITY_NONE",
+            "_WINDQUALITY_FASTEST",
+            "_WINDQUALITY_FAST",
+            "_WINDQUALITY_BETTER",
+            "_WINDQUALITY_BEST",
+            "_WINDQUALITY_PALM"
+        };
+
+        /// <summary>
+        /// Creates a material upgrader with only the renames in common between HD and Universal.
+        /// </summary>
+        /// <param name="sourceShaderName">Original ST8 shader name.</param>
+        /// <param name="destShaderName">New ST8 shader name.</param>
+        /// <param name="finalizer">A delegate that postprocesses the material as needed by the render pipeline.</param>
+        public SpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName, MaterialFinalizer finalizer = null)
+        {
+            RenameShader(sourceShaderName, destShaderName, finalizer);
+            RenameFloat("_WindQuality", "_WINDQUALITY");
+            RenameFloat("_TwoSided", "_CullMode"); // Currently only used in HD. Update this once URP per-material cullmode is enabled via shadergraph. 
+        }
+
+        private static void ImportNewSpeedTree8Material(Material mat, int windQuality, bool isBillboard)
+        {
+            int cullmode = 0;
+            mat.SetFloat("_WINDQUALITY", windQuality);
+            if (isBillboard)
+            {
+                mat.EnableKeyword("EFFECT_BILLBOARD");
+                cullmode = 2;
+            }
+            if (mat.HasProperty("_CullMode"))
+                mat.SetFloat("_CullMode", cullmode);
+        }
+
+        /// <summary>
+        /// Postprocess materials when importing a SpeedTree8 asset. Call from OnPostprocessSpeedTree in a MaterialPostprocessor.
+        /// </summary>
+        /// <param name="speedtree">Game object for the SpeedTree asset being imported.</param>
```

Comment:
"for" is unclear here. 
Would this rewrite be correct?
`The GameObject Unity creates from this imported SpeedTree.`